CONTRIBUTING.md

```diff
@@ -29,21 +29,47 @@ Qt:
   will (very likely) be **ignored**. :fearful:
 
 
-# Opening a pull request
+# How to start a pull request
 
-### Must read
+## Must read
+* Before you start contributing, first decide for a specific topic. Pull requests, which are spanning multiple topics (e.g. "general qTox code cleanup") or introduce fundamental architectural changes are rare and require additional attention and maintenance.
 * Use [**commit message format**](#commit-message-format).
 * Read our [**coding guidelines**](#coding-guidelines).
 * Keep the title **short** and provide a **clear** description about what your
   pull request does.
 * Provide **screenshots** for UI related changes.
-* Keep your git commit history **clean** and **precise**. Commits like
-  `xxx fixup` should not appear.
+* Keep your git commit history **clean** and **precise** by continuously rebasing/amending your PR. Commits like `xxx fixup` are not needed and rejected during review.
 * If your commit fixes a reported issue (for example #4134), add the following
   message to the commit `Fixes #4134.`.
   [Here is an example](https://github.com/tux3/qTox/commit/87160526d5bafcee7869d6741a06045e13d731d5).
 
-### Good to know
+## How to avoid merge conflicts
+If you experience big amounts of conflicts, they are indicating a problem in your communication and appear e.g. on PR's, that try to "fix everything". So the best way is to completely avoid "hard-to-solve" merge conflicts by concentrating on smaller related chunks of code. Anyway, if you run into such situation, here's a "Rule of Thumb" list to get out of it:
+
+1. Split your commit history into topic related chunks (by rebasing them).
+2. Split "API" and "UI" (widget related) changes into separate commits.
+3. Probably split PR into several smaller ones.
+
+In addition it helps to regularly keep rebasing on the main repository's recent master. You can do this by setting up a second Git remote as follows:
+
+~~~bash
+# Add a Git remote for the qTox main repository:
+git remote add main https://github.com/tux3/qTox.git
+
+# If not on PR branch, check it out:
+git checkout my/pr-branch
+
+# Now fetch master ALWAYS from main repo
+git fetch main master:master
+
+# Last, rebase PR branch onto master…
+git rebase -i master
+
+# …and, if everything's clear, force push to YOUR repo (your "origin" Git remote)
+git push -f
```

Comment:
**I would never recommend one to do so and I won't change that here!** As I tried to explain, there's nothing wrong with this workflow and I use it exactly that way every day! IOW: When you're in the need to force push, you're in the process of cleaning up (rebasing/amending) commits for an existing upstream branch. Just a few days ago initramfs ran into this problem and accidentally pushed his detached HEAD during an in-progress rebase. By following Git's deafault workflow and set the upstream branch right from the start, when creating a PR (without any fancy `--force` option), this is completely safe. Understand? :)

So, when writing the docs, it could change to something like this:

# How to open/setup a PR

1. Fork the qTox repository on Github to your existing account.
2. Open a Terminal and do the following steps:
```bash
# Go to your projects directory, where the qTox directory will be created:
cd /to/the/directory

# Clone the forked repo:
git clone git@github.com:<YOUR_USER>/qTox.git

# Add the upstream repository to be able to fetch from:
git remote add https://github.com/qTox/qTox.git

# Point the local "master" branch to the "upstream" repository
git branch master --set-upstream-to=upstream/master
```

You're now all setup to create your first pull request:

Still in Terminal, do the following steps: 

```bash
# Fetch from the "upstream" repository:
git fetch upstream master:master

# Checkout a local branch on up-to-date "master" and give it a sane name, e.g.:
git checkout -b feat/brandnew-feature master
```

Now do your changes and commit them by your heart's desire. When you think you're ready to push for the first time, do the following:

```bash
# Push to the new upstream branch and link it for synchronization
git push -u origin feat/brandnew-feature

# From now on, you can simply…
git push
# ...to your brand new pull request. 
```